Packs/EWS/Integrations/EWSv2/EWSv2.py

```diff
@@ -917,23 +917,21 @@ def fetch_last_emails(account, folder_name='Inbox', since_datetime=None, exclude
     qs = qs.filter().order_by('datetime_received')
 
     result = []
+    counter = 0
     for item in qs:
+        counter += 1
         try:
-            if isinstance(item, Message):
+            if isinstance(item, Message) and item.message_id not in exclude_ids:
                 result.append(item)
                 if len(result) >= MAX_FETCH:
                     break
         except ValueError as exc:
             future_utils.raise_from(ValueError(
                 'Got an error when pulling incidents. You might be using the wrong exchange version.'
             ), exc)
-
-    if exclude_ids and len(exclude_ids) > 0:
-        exclude_ids = set(exclude_ids)
-        result = [x for x in result if x.message_id not in exclude_ids]
+    demisto.debug(f'EWS V2 - Got total of {counter} from ews query. {len(result)} results not excluded. ')
```

Comment:
```suggestion
    demisto.debug('EWS V2 - Got total of {} from ews query. {} results not excluded. '.,fomat(counter, len(result)))
```
EWSv2 is python2 code